### PR TITLE
Allow to explicitly pass the codecov token; bump codecov action to version 4

### PR DESCRIPTION
**Pull request by felixfontein**

### CODE REVIEW

**Review by webknjaz — APPROVED** [review]

**Comment by webknjaz** [comment]
@felixfontein thanks! This needs rebasing.

**Comment by webknjaz** [comment]
pre-commit.ci run

**Comment by webknjaz** [comment]
pre-commit.ci run

**Comment by felixfontein** [comment]
@webknjaz thanks for reviewing and merging!

File: action.yml [hunk 1]
```diff
@@ -682,6 +685,8 @@ runs:
           inputs.testing-type
         }}=*.xml
       flags: ${{ inputs.testing-type }}
+      token: ${{ inputs.codecov-token }}
+      working-directory: ${{ steps.collection-metadata.outputs.checkout-path }}
```
Comment (webknjaz) [inline action.yml:689]: Looks like this ended up being overindented..

Comment (webknjaz) [reply action.yml:689]: Ah, nevermind

**Pull request merged**